Mte90/GlotDict
# Whats new

What's new and Welcome...
I first made a small modification on the GlotDict menu to make it more visible: enlarging the icon, and adding an arrow that turns around when you open the settings.
![menu](https://user-images.githubusercontent.com/17084006/132954616-e045ee94-b631-4f8b-b9fb-20b113748b4a.jpg)

Then I added tabs to the parameters to be able to add the notion of What's new and Welcome.
As the tabs could not be seen on the white background, I integrated them into the upper blue part which allows them to be clearly distinguished.
Since Vlad's latest modifications have reduced the length of the lines, I was able to modify the responsive breakpoint so that the parameters on 2 columns are also displayed on the notebooks.
![notebook](https://user-images.githubusercontent.com/17084006/132954703-d7c8c0b1-a4a5-4bf4-a7f0-45ead47952a9.jpg)

The principle is simple:
When GlotDict is installed for the first time, the 2nd tab opens automatically and the welcome message is displayed. I thank Vlad for writing it.
![install](https://user-images.githubusercontent.com/17084006/132954739-07df04e4-722a-4026-873c-0a61470a8b1b.jpg)

A cross to close the settings performs the same action as clicking on the menu: it is there for newbies who may not find the menu the first time.

When GlotDict is updated, the 2nd tab opens automatically and the last part of the changelog is displayed. This part is automatically retrieved from the real changelog which will save us some work.

![whatsnew](https://user-images.githubusercontent.com/17084006/132954817-f21bbc83-4bb3-4ac4-aa73-c36e8120ba7d.jpg)

How it works:
The only real content script (not injected into the page) "init.js" retrieves the changelog data, then sends a request via sendMessage to background.js then inject scripts. The order is mandatory. Background.js is declared as a background script, and it is the only one that can access the required data. It sends them back to init.js which is responsible for saving them in localStorage.
Glodict.js declares a global gd_extension which it populates from localStorage and displays or not the parameters depending on the extension update or installation.

How to test it:
A quick test would be to modify the values in the localStorage, but this is not the safest since they arrive later.
The real test is to remove the extension and reinstall it to test the installation, and change the version number in the manifest to test the update. When doing successive tests, you should especially think about deleting the gd_extension_status value in the localStorage.
